src/components/select/select-list/select-list.component.js

```diff
@@ -0,0 +1,170 @@
+import React, {
+  useEffect,
+  useState,
+  useCallback
+} from 'react';
+import PropTypes from 'prop-types';
+import StyledSelectList from './select-list.style';
+import Events from '../../../utils/helpers/events/events';
+import updateListScrollTop from './update-list-scroll';
+import getNextChildByText from '../utils/get-next-child-by-text';
+import Portal from '../../portal/portal';
+import { getNextIndexByKey } from '../utils/get-next-child-by-key-code';
+
+const SelectList = React.forwardRef(({
+  id,
+  children,
+  onSelect,
+  onSelectListClose,
+  filterText,
+  anchorElement,
+  highlightedValue
+}, listRef) => {
+  const [currentOptionsListIndex, setCurrentOptionsListIndex] = useState(-1);
+
+  const highlightItem = useCallback((newIndex, optionList) => {
+    const { text, value } = optionList[newIndex].props;
+
+    setCurrentOptionsListIndex(newIndex);
+    updateListScrollTop(newIndex, listRef.current);
+    onSelect({ text, value, selectionType: 'navigationKey' });
+  }, [listRef, onSelect]);
+
+  const handleGlobalKeydown = useCallback((event) => {
+    const optionList = React.Children.toArray(children);
+    const lastIndex = optionList.length - 1;
+
+    if (Events.isTabKey(event) || Events.isEscKey(event)) {
+      onSelectListClose();
+    } else if (Events.isEnterKey(event)) {
+      const currentOption = optionList[currentOptionsListIndex];
+
+      if (!currentOption) {
+        onSelectListClose();
+
+        return;
+      }
+
+      const { text, value } = currentOption.props;
+
+      onSelect({ text, value, selectionType: 'enterKey' });
+    } else if (isNavigationKey(event)) {
+      const nextIndex = getNextIndexByKey(event.which, currentOptionsListIndex, lastIndex);
+
+      highlightItem(nextIndex, optionList);
+    }
+  }, [currentOptionsListIndex, children, onSelectListClose, onSelect, highlightItem]);
+
+  const repositionList = useCallback(() => {
+    if (anchorElement) {
+      const inputBoundingRect = anchorElement.getBoundingClientRect();
+      const top = `${inputBoundingRect.top + inputBoundingRect.height + window.pageYOffset}px`;
+      const width = `${inputBoundingRect.width}px`;
+      const left = `${inputBoundingRect.left}px`;
+
+      listRef.current.setAttribute('style', `top: ${top}; width: ${width}; left: ${left}`);
```

Comment:
I've noticed if you resize the window, the select-list is positioned without respect to scroll position.

![image](https://user-images.githubusercontent.com/2328042/86144122-2a6ca480-baed-11ea-929b-2b21338efe31.png)

Reply:
I cannot reproduce it, could you provide the steps you have taken?

Reply:
• Go to https://codesandbox.io/s/select-review-vbx73
• Scroll the preview pane to the right so it looks like this
![image](https://user-images.githubusercontent.com/2328042/86596031-81f98d00-bf91-11ea-9945-269b5461751d.png)
• Open the dropdown using the mouse
![image](https://user-images.githubusercontent.com/2328042/86596188-b9683980-bf91-11ea-9de6-8f5eca1ae0c3.png)

Reply:
Looks like it's an issue specific to content being embedded in an iframe in the Codesandbox

it affects every carbon component rendered in a portal, for example the Date Component:

![Screenshot 2020-07-06 at 16 38 26](https://user-images.githubusercontent.com/22885392/86605929-9c3e6580-bfa7-11ea-9294-9bbfb4c8dde5.png)

![Screenshot 2020-07-06 at 16 38 16](https://user-images.githubusercontent.com/22885392/86605949-a2ccdd00-bfa7-11ea-9834-22ccfd6ba007.png)

Reply:
OK, please can you create a JIRA ticket and we'll fix it separately.

Reply:
Issue was fixed for this component.